File: CultureLinkCRM/CultureLinkCRM.Web/Modules/ServerTypes/CrmDB/CityRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface CityRow {
    CityId?: number;
    Name?: string;
}

export abstract class CityRow {
    static readonly idProperty = 'CityId';
    static readonly nameProperty = 'Name';
    static readonly localTextPrefix = 'CrmDB.City';
    static readonly lookupKey = 'CrmDB.City';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<CityRow>('CrmDB.City') }
    static async getLookupAsync() { return getLookupAsync<CityRow>('CrmDB.City') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<CityRow>();
}

File: CultureLinkCRM/CultureLinkCRM.Web/Modules/CrmDB/City/CityRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace CultureLinkCRM.CrmDB;

[ConnectionKey("Default"), Module("CrmDB"), TableName("City")]
[DisplayName("City"), InstanceName("City")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]
[LookupScript]
public sealed class CityRow : Row<CityRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("City Id"), PrimaryKey, IdProperty]
    public int? CityId { get => fields.CityId[this]; set => fields.CityId[this] = value; }

    [DisplayName("Name"), QuickSearch, NameProperty]
    public string Name { get => fields.Name[this]; set => fields.Name[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field CityId;
        public StringField Name;

    }
}